Farm Mode Data Binding + Offline Sync

johnswyoon opened the pull request:
# Farm Mode Data Binding

**What Was Changed:**
- Connected Firestore to Farm Mode UI
    - Adding new item adds to Firestore
    - Adding existing item updates quantity in Firestore
- SYNC button for offline mode
    - If offline and user tries to update, the data will be stored to SYNC later (when reconnected)
    - User should press SYNC when they re-connect to internet to send data to Firestore

-------

**Testing Steps:**
- Go to offline mode (turn off wifi)
- Add products to table and press update
- Pop will say press SYNC when connected to wifi
    - Table will clear, but the products will be saved
- Turn on wifi
- Press Sync
- Go to Home Mode's inventory -> you should see ur new products

---------------

**Current Issues:**
- Before you press sync/update, if you go to another page everything will reset
    - This is because when you leave and return to a fragment, the onCreateView is triggered and resets all variables
- Random crashes when in Farm Mode (usually switching from Farm Mode -> another mode)

---------------
Added a popup explaining to user that offline mode can't send data to Firestore:

![Screenshot 2023-07-22 at 2 03 31 AM](https://github.com/JacobRintjema/ece452project/assets/55363559/825ca37a-4906-4484-9ed4-b154feb2366e)

johnswyoon committed:
Merge branch 'master' of https://github.com/JacobRintjema/ece452project into farmmode-data-binding

johnswyoon committed:
clear table

johnswyoon committed:
working with wifi

johnswyoon committed:
working offline sync

johnswyoon committed:
Merge branch 'master' of https://github.com/JacobRintjema/ece452project into farmmode-data-binding

johnswyoon committed:
adding same product but updating quantity not working

johnswyoon committed:
progress for addProduct

johnswyoon committed:
working update quantity for same item name

JacobRintjema commented:
![Screenshot 2023-07-23 at 6 11 03 PM](https://github.com/JacobRintjema/ece452project/assets/55416063/8df5e8b4-2733-468e-bed1-1e89379e8445)
Followed the steps and works as intended, just wondering if we need the "sync" button or if that should be done automatically for the user when they regain connection?

JacobRintjema commented:
Also in line with the "Current Issues" that you wrote, why can't we just save/cache the products when they are made to being with, rather than waiting until the user clicks Update?

This would get rid of the problem where the page resets / products are lost if you don't click Update/Sync

johnswyoon committed:
finished farm mode

johnswyoon commented:
- Fixed merge conflicts
- clear inputs when pressing add
- fixed UI to match jocelyns

Decided to not implement @JacobRintjema changes due to time

johnswyoon committed:
Update Product.kt
